### DISCUSSION TITLE
viewport height - as a percentage of the screen?

### DISCUSSION THREAD

**Discussion by benkoshy** [tailwindlabs/tailwindcss#8903] [Help] [answered]
Hi there

Was wondering if there is a way to get a height of 85% of a viewport height. something like this?

```html
<div class="h-screen-85"> </div>
```

https://tailwindcss.com/docs/height#viewport-height

The problem I am trying to solve is to ensure my footer is minimally at the bottom of the screen, even if the parent elements are much shorter in height:

```html
<main> ... </main>

<footer> ... </footer> <-- must never be halfway on the screen -->
```

I would never ever want my footer to be half way up the page, as it is now:

So I wanna set a minimum height of the main body elements above the footer.

![image](https://user-images.githubusercontent.com/15097447/179946474-d43a045d-9f71-4393-80fe-7c034bba6f3f.png)

**Comment by ichenglin** [comment, marked as answer]
You could **apply css values** in tailwind with **[Arbitrary Properties](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties)**,
which is extremely useful in your situation.

Here's an example of applying **`85% height`** to an **`div element`**
```html
<div class="w-full h-[85%]"></div>
```

**Reply by ichenglin** [reply]
An **additional update** to the solution, credit to **[arisac](https://github.com/arisac)** for commenting below,
which reminds me something I forgot to include.

| Unit | Definition |
|-|-|
|%|Percent of **parent element**|
|vw|Percent of view **port width**|
|vh|Percent of view **port height**|

In my solution I assume you're **applying it to the root element** (right inside body),
which **`h-[85%]`** would **perform the same** as **`h-[85vh]`**.

On the other hand, if you're **NOT applying it to the root element**,
you should use **`h-[85vh]`** instead.

**Reply by benkoshy** [reply]
Please correct me if I'm wrong - what you are proposing is (see below)?

```html
<main class="h-[85vh]"> ... </main>
<footer> ... </footer> <-- the remaining 15% -->
```
Or alternatively:
```html
<main>
   <div class="w-full h-[85%]"></div>
   <footer> ... </footer>
<main>
```

Am I misunderstanding?

(Icheng and Aris - very much appreciate your pointers (@ichenglin  @arisac))

**Reply by arisac** [reply]
use `h-[85vh]` unit if you mean exactly 85% of the viewport(browser view height). Always use `vh/vw` unit if you want to relative to the viewport.

and for your use case, you should set min-height for the content. play for yourself 👉 https://play.tailwindcss.com/Vn0wYiJbsA

learn css units 👉 https://www.w3schools.com/cssref/css_units.asp

**Reply by benkoshy** [reply]
Sweet. thanks for the pointer. I will take your advice.

**Comment by arisac** [comment]
for 85% screen height: `h-[85vh]`

see demo https://play.tailwindcss.com/fzbOWcbdJ8

**Reply by ichenglin** [reply]
There's an **error** in the solution you provided,
you should be using **`vh`** instead of **`vw`** for view port height.
